### PR TITLE
Samples unit test data

### PR DESCRIPTION
**Pull request by mihail911**
Added data files and script for generation as requested. Please review @BoltzmannBrain

### CODE REVIEW

File: fluent/utils/unit_test_generation.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    @return  NONE
+    """
+    with open(inputData, "r") as fRead:
+      with open("sample_reviews_data_training_scramble.csv", "w") as fWrite:
+        fileReader = csv.DictReader(fRead)
+        fileWriter = csv.DictWriter(fWrite, fieldnames=fileReader.fieldnames)
+        fileWriter.writeheader()
+        for sample in fileReader:
+          response = sample["Response"]
+          tokens = processor.tokenize(response,
```
Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: don't preprocess the samples

File: data/sample_reviews_unit/sample_reviews_data_training_semantics.csv [hunk 2]
```diff
@@ -0,0 +1 @@
...
27,,"We should each be given a personal whiteboard at our work space. Further, we should have webcames in all the conference rooms.",tools,tech,
28,,Change the arrangements of a conference room so that it is actually a room for sleeping.,productivity,,
29,,"It would be nice to have lecture series, guest lectures with videos or lectures that would give.",productivity,entertainment,
30,,Higher quality chairs.,furniture,,
31,,My current working space is the ideal one where I could potentially find myself. It could be better if we had more drinks and movie nights.,positive,entertainment,food
32,,Healthier snacks and bringing us dinner when we stay after hours. It would also be nice to have time for workouts during lunch as well as a gym membership at the establishment down the street.,work/life balance,food,
33,,Our surveys should be fewer in number.,not helpful,,
34,,"We should have flying vehicles, video games, a group for discussing literature, and better beer. ",entertainment,tech,food
35,,"The office is perfect. All our accommodations, including food, technology, and toiletries, are the best they could possibly be.",positive,,
```
Comment (BoltzmannBrain) [inline data/sample_reviews_unit/sample_reviews_data_training_semantics.csv]: change 14 to "... company rewards 10,000 dollars to..."

File: fluent/utils/unit_test_generation.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    shuffled.
+
+    @param  (str)               Path to input original samples data file
+    @param  (TextPreprocess)    Processor to perform some text cleanup.
+
+    @return  NONE
+    """
+    with open(inputData, "r") as fRead:
+      with open("sample_reviews_data_training_scramble.csv", "w") as fWrite:
+        fileReader = csv.DictReader(fRead)
```
Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: use csv_helper.py for general csv operations

File: fluent/utils/unit_test_generation.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+def generateRandomFile(inputData, processor):
+    """
+    Generates a samples data file with all of the words per sample randomly
+    shuffled.
+
+    @param  (str)               Path to input original samples data file
+    @param  (TextPreprocess)    Processor to perform some text cleanup.
+
+    @return  NONE
```
Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: don't need to specify this; all Python methods return None if there is no return

File: fluent/utils/unit_test_generation.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+          shuffledResponse = " ".join(tokens)
+          outputDict = {"QID": sample["QID"], "QuestionText":
+              sample["QuestionText"], "Response": shuffledResponse,
+              "Classification 1": sample["Classification 1"],
+              "Classification 2": sample["Classification 2"],
+              "Classification 3": sample["Classification 3"]}
+          fileWriter.writerow(outputDict)
+
+
+def generateReversedFile(inputData, processor):
```
Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: This function looks almost identical to `generateRandomFile`... wouldn't it be cleaner code to have one function that is called twice?

File: fluent/utils/unit_test_generation.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+# ----------------------------------------------------------------------
+"""
+This file contains CSV utility functions to use with nupic.fluent experiments.
+"""
+
+import argparse
+import csv
+import os
+import random
+import sys
```
Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: ?? This isn't used anywhere.

**Comment by BoltzmannBrain** [comment]
@mihail911 done w/ review, left some comments. Don't tokenize and then scramble/reverse the samples, keep the words in their original forms. We don't want contractions to be split and then have a random "t" floating around in the sample.

File: fluent/utils/unit_test_generation.py (hunk 1 above)
Comment (mihail911) [inline fluent/utils/unit_test_generation.py]: so you want to keep the "[identifier deleted]" in the samples? Should this be treated as a single token or is it alright if the randomization separates the tokens "[identifier" and "deleted]"

Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: "[identifier deleted]" is one word, keep it as such

**Comment by mihail911** [comment]
Please review @BoltzmannBrain

File: fluent/utils/unit_test_generation.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+This file contains CSV utility functions to use with nupic.fluent experiments.
+"""
+
+import argparse
+import os
+import random
+
+from fluent.utils.csv_helper import readCSV, writeCSV
+
+dataDir = "/Users/numenta/Documents/nupic.fluent/data/sample_reviews_unit"
```
Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: Don't hardcode a path; it's likely different on every machine. Just write to the same directory as the input data.

File: fluent/utils/unit_test_generation.py [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    reversed.
+
+    @param  (str)               Path to input original samples data file
+    @param  (TextPreprocess)    Processor to perform some text cleanup.
+
+    """
+    if type == "scrambled":
+      fileName = "sample_reviews_data_training_scramble.csv"
+    elif type == "reversed":
+      fileName = "sample_reviews_data_training_reverse.csv"
```
Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: What if my input data isn't named "sample_reviews_data_training.csv"?? Better to use string logic...

    import string
    filename = string.join(inputData.split(".")[:-1], ".") + "_" + type + ".csv"

**Comment by BoltzmannBrain** [comment]
@mihail911 data looks good, but there is some unnecessary messing around with path/file naming; I left some comments.

**Comment by BoltzmannBrain** [comment]
@mihail911 where are the files?

File: fluent/utils/unit_test_generation.py [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+import argparse
+import os
+import random
+import string
+
+from fluent.utils.csv_helper import readCSV, writeCSV
+
+
+def cleanTokens(tokens):
```
Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: three blank lines before top level function

File: fluent/utils/unit_test_generation.py [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+                                one token instead of two.
+    """
+    cleanTokens = []
+    for token in tokens:
+      if token == "[identifier":
+        cleanTokens.append("[identifier deleted]")
+      elif token != "deleted]":
+        cleanTokens.append(token)
+    return cleanTokens
+
```
Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: two blank lines between functions

File: fluent/utils/unit_test_generation.py [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+        response = " ".join(tokens)
+      elif type == "reversed":
+        response = " ".join(tokens[::-1])
+
+      dataToWrite = [sample[0], "", response]
+      dataToWrite.extend(sample[1][1])
+      data.append(dataToWrite)
+
+    writeCSV(data, headers, os.path.join(outputDataDir, fileName))
+
```
Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: 2 lines...

File: fluent/utils/unit_test_generation.py [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+def generateDataFile(inputData, type):
+    """
+    Generates a samples data file with all of the words in the sample
+    reversed.
+
+    @param  (str)               Path to input original samples data file
+    @param  (TextPreprocess)    Processor to perform some text cleanup.
+
+    """
+    outputDataDir = os.path.dirname(inputData)
```
Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: this is marginally better than hardcoding the output path... why wouldn't you just make this an arg like the input path?

File: fluent/utils/unit_test_generation.py [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+
+      dataToWrite = [sample[0], "", response]
+      dataToWrite.extend(sample[1][1])
+      data.append(dataToWrite)
+
+    writeCSV(data, headers, os.path.join(outputDataDir, fileName))
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("inputData",
```
Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: add a similar argument for `outputDataDir`, with a default value `test_data'

Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: add the param `required=True`

**Comment by BoltzmannBrain** [comment]
@mihail911 left some more comments. Please address them and push the resulting data files.

**Comment by mihail911** [comment]
Made updates. Please review @BoltzmannBrain

**Comment by BoltzmannBrain** [comment]
:+1:

File: fluent/utils/unit_test_generation.py [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+"""
+This file contains CSV utility functions to use with nupic.fluent experiments.
```
Comment (BoltzmannBrain) [inline fluent/utils/unit_test_generation.py]: bad copy and paste

**Pull request merged**